### PR TITLE
magnetic field specific refinement

### PR DESCRIPTION
**Pull request by nicolasaunai**
add refinement specific to magnetic field
on top of #715

### CODE REVIEW

File: src/amr/data/field/coarsening/magnetic_field_coarsener.hpp [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+                coarseField(coarseIndex[dirX], coarseIndex[dirY])
+                    = 0.25
+                      * (fineField(fineStartIndex[dirX], fineStartIndex[dirY])
+                         + fineField(fineStartIndex[dirX] + 1, fineStartIndex[dirY])
+                         + fineField(fineStartIndex[dirX], fineStartIndex[dirY] + 1)
+                         + fineField(fineStartIndex[dirX] + 1, fineStartIndex[dirY] + 1));
+            }
+            else
+            {
+                {
```
Comment (PhilipDeegan) [inline src/amr/data/field/coarsening/magnetic_field_coarsener.hpp:115]: redundant braces

File: src/amr/data/field/field_variable_fill_pattern.hpp [hunk 2]
```diff
@@ -156,6 +156,9 @@ class FieldFillPattern : public SAMRAI::xfer::VariableFillPattern
 
         SAMRAI::hier::BoxContainer overlap_boxes(fill_boxes);
         overlap_boxes.intersectBoxes(data_box);
+        //        overlap_boxes.removeIntersections(patch_box);
```
Comment (PhilipDeegan) [inline src/amr/data/field/field_variable_fill_pattern.hpp:159]: rm?

File: src/amr/data/field/patch_level_ghost_fill_pattern.cpp [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+    NULL_USE(fill_ghost_width);
+    NULL_USE(dst_fill_boxes_on_src_proc);
+    if (!needsToCommunicateDestinationFillBoxes())
+    {
+        TBOX_ERROR("PatchLevelGhostFillPattern cannot compute destination:\n"
+                   << "fill boxes on the source processor.\n");
+    }
+}
+
+bool PatchLevelGhostFillPattern::needsToCommunicateDestinationFillBoxes() const
```
Comment (PhilipDeegan) [inline src/amr/data/field/patch_level_ghost_fill_pattern.cpp:134]: simple function implementations (one liners) can go in headers where they're declared I think

**Comment by nicolasaunai** [comment]
obsolete

**Pull request closed without merging**